_posts/2020-12-24-how-to-start-with-foreman

```diff
@@ -0,0 +1,399 @@
+---
+layout: post
+title: How to start with Foreman
+date: 2020-12-24 17:00:00
+author: Marek Hulan
+tags:
+- foreman
+- community
+- tutorial
+- howto
+- beginners
+---
+
+This post targets primarily users new to Foreman. It can be interesting for those,
+who would like to start using Foreman but feel lost or who just installed Foreman
+and would like to investigate its capabilities. We'll assume a fresh Foreman
+installation without any plugin and with the default configuration. The primary focus 
+will be on on filling in the data about the infrastructure. If you're searching for
+installation instructions, please take a look at [this chapter](https://theforeman.org/manuals/2.3/index.html#3.InstallingForeman)
+of the Foreman manual. This post assumes installation using the foreman-installer.
+
+<!--more-->
+
+Foreman allows to define many objects, which serve different purposes. But the
+heart of everything is the Host object, It glues all other objects together.
+Host represents some machine. It can be a device on network or simply something,
+that you want to install operating system. It's typically a thing you want to manage
+or automate some tasks around it.
+In virtualization terminology, it can be host but also a guest. It can be a baremetal
+machine, it can be VM running on some hypervisor or a VM somewhere in the cloud.
+While Foreman does not primarily target these, it can also be a router, IoT sensor,
+laptop, container.
+
+Regardless of what kind of machine the host is, Foreman calls that simply the Host. Hosts have name.
+They also must belong to some organization and location. This is the bare minimum
```

Comment:
No, they don't have to. But then you get funny bugs like https://bugzilla.redhat.com/show_bug.cgi?id=1909657 ;)

Reply:
these weird things will be coverd in the next blog called `why you should avoid taxonomies` :-)

I believe while there are ways to achieve this, the managed host is then considered invalid. DB is in inconsistent state then. And it's definitely not something I want to recommend to people starting with Foreman :-)

Reply:
The "funny" thing is, the original foreman host (id 1) has no taxonomies attached.

Reply:
and therefore most user will never see it :-) but yeah, definitely something to finally fix